templates/core/core-secret.yaml

```diff
@@ -1,18 +1,26 @@
+{{- $secret := include "harbor.core" . }}
 apiVersion: v1
 kind: Secret
 metadata:
-  name: {{ template "harbor.core" . }}
+  name: {{ $secret }}
   labels:
 {{ include "harbor.labels" . | indent 4 }}
 type: Opaque
 data:
-  secretKey: {{ .Values.secretKey | b64enc | quote }}
+{{- if .Release.IsInstall }}
+  secretKey: {{ .Values.core.secretKey | default (randAlphaNum 16) | b64enc | quote }}
   secret: {{ .Values.core.secret | default (randAlphaNum 16) | b64enc | quote }}
+  HARBOR_ADMIN_PASSWORD: {{ .Values.harborAdminPassword | default (randAlphaNum 16) | b64enc | quote }}
+  CSRF_KEY: {{ .Values.core.xsrfKey | default (randAlphaNum 32) | b64enc | quote }}
+{{- else }}
+  secretKey: {{ (lookup "v1" "Secret" .Release.Namespace $secret).data.secretKey }}
+  secret: {{ (lookup "v1" "Secret" .Release.Namespace $secret).data.secret }}
+  HARBOR_ADMIN_PASSWORD: {{ (lookup "v1" "Secret" .Release.Namespace $secret).data.HARBOR_ADMIN_PASSWORD }}
```

Comment:
I found this in the [helm documentation](https://helm.sh/docs/chart_template_guide/functions_and_pipelines/):

_Keep in mind that Helm is not supposed to contact the Kubernetes API Server during a helm template or a helm install|update|delete|rollback --dry-run, so the lookup function will return an empty list (i.e. dict) in such a case._

Reply:
So ... ? `--dry-run` is not a problem, I assume

Reply:
Hm, but that `lookup` during template call is probably the reason why `lint` fails...